hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/KeyManagerImpl.java

```diff
@@ -2134,11 +2140,13 @@ public List<OmKeyInfo> getPendingDeletionSubFiles(long volumeId,
         fileInfo.setKeyName(fullKeyPath);
 
         files.add(fileInfo);
+        remainingBufLimit -= objectSerializedSize;
+        consumedSize += objectSerializedSize;
```

Comment:
First will check if crossing limit, if crosses limit, reject the current one and return. Otherwise in boundary cases, it can fail.

Reply:
Done.